src/libtorch.cc

```diff
@@ -1341,8 +1523,9 @@ ModelInstanceState::Execute(
 
     // NV-Fuser. No change is made unless parameter is explicitly set.
     if (std::get<0>(model_state_->EnabledNvfuserPair())) {
-      if (std::get<1>(model_state_->EnabledNvfuserPair()) &&
-          (device_.type() != torch::kCPU)) {
+      if ((std::get<1>(model_state_->EnabledNvfuserPair()) &&
+           (device_.type() != torch::kCPU)) ||
+          (Kind() == TRITONSERVER_INSTANCEGROUPKIND_MODEL)) {
```

Comment:
Would this work if the model is loaded with KIND_MODEL but runs on CPU and the container is loaded without --gpus flag?

Reply:
Good catch. I think we need to add the check for available devices here.

Reply:
Fixed the condition.